src/custom-elements/src/components/search-input/search-input.tsx

```diff
@@ -0,0 +1,161 @@
+   */
+  @Prop({ reflect: true }) preventValueMutation = false;
+  /**
+   * To define -hover, -focus statuses
+   */
+  @Prop() _status: string;
+
+  @Element() el: HTMLElement;
+
+  @State() _cleanButtonVisible = (this.value && !this.disabled) ? true : false;
+
+  /**
+   * Triggered when an alteration to the element's value is committed by the user
+   */
+  @Event({ eventName: 'chiChange' }) eventChange: EventEmitter<string>;
+  /**
+   * Triggered when the user changed the element's value but did not commit the change yet
+   */
+  @Event({ eventName: 'chiInput' }) eventInput: EventEmitter<string>;
+  /**
+   * Triggered when the user sets focus on the element.
+   */
+  @Event({ eventName: 'chiFocus' }) eventFocus: EventEmitter;
+  /**
+   * Triggered when the element has lost focus.
+   */
+  @Event({ eventName: 'chiBlur' }) eventBlur: EventEmitter;
+  /**
+   * Triggered when the user clicked the X button.
+   */
+  @Event({ eventName: 'chiClean' }) eventClean: EventEmitter;
+
+  @Watch('size')
+  sizeValidation(newValue: TextInputSizes) {
+    const validValues = TEXT_INPUT_SIZES.join(', ');
+
+    if (newValue && !TEXT_INPUT_SIZES.includes(newValue)) {
+      throw new Error(`${newValue} is not a valid size for input. If provided, valid values are: ${validValues}. `);
+    }
+  }
+
+  _handleValueInput(valueChange: Event) {
+    const newValue = (valueChange.target as HTMLInputElement).value;
+
+    if (!this.preventValueMutation) {
+      this.value = newValue;
+    }
+
+    if (newValue === "") {
+      this._cleanButtonVisible = false;
+    } else {
+      this._cleanButtonVisible = true;
+    }
+    this.eventInput.emit(newValue);
+  }
+
+  _handleValueChange(valueChange: Event) {
+    const newValue = (valueChange.target as HTMLInputElement).value;
+
+    this.eventChange.emit(newValue);
+  }
+
+  _cleanInput() {
+    const input = this.el.querySelector('input[type=search]');
+
+    this.value = '';
+    (input as HTMLInputElement).focus();
+    this._cleanButtonVisible = false;
+    this.eventClean.emit();
+  }
+
+  componentWillLoad() {
+    this.sizeValidation(this.size);
+  }
+
+  render() {
+    const searchInputElement = <input
+      type="search"
+      class={`
+        chi-input chi-search__input
+        ${this.size ? `-${this.size}` : ''}
+        ${this._status ? `-${this._status}` : ''}
+      `}
+      placeholder={this.placeholder || ''}
+      value={this.value}
+      name={this.name || ''}
+      disabled={this.disabled}
+      id={this.el.id ? `${this.el.id}-control` : null}
+      onFocus={() => this.eventFocus.emit()}
+      onBlur={() => this.eventBlur.emit()}
+      onInput={(ev) => this._handleValueInput(ev)}
+      onChange={(ev) => this._handleValueChange(ev)}
+      autocomplete="off"
+      aria-label="search input"
+    />;
+
+    const searchXIcon = this._cleanButtonVisible ?
+      <button class={`
+        chi-button -icon -close
+        ${this.size === 'sm' ? '-sm' : ''}
```

Comment:
I would suggest using the small X for all sizes. The large X served a different purpose which the UX team later omitted from the spec.